### PR TITLE
Avoid code repetition in Profile Edit/Creation

### PR DESCRIPTION
**Pull request by bessmarche**
Unify the jsp pages and servlets for user profile creation and editing.
There is now only one JSP page: **profile-creation.jsp** that shows slight modification to the user depending of whether the user is logged in or not.
The **ProfileCreationServlet.java** now handles both cases, by checking is the user that submitted the request is already log in (hence they are **editing the profile**) or if it is not logged in (hence they are **creating the profile**)

### CODE REVIEW

**Review by donpaul999 — APPROVED** [review]

**Review by donpaul999 — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/ProfileCreationServlet.java [hunk 1]
```diff
@@ -38,33 +38,72 @@
...
    */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response)
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
     throws ServletException, IOException {
-    String username = request.getParameter("username");
-    String bio = request.getParameter("bio");
 
     HttpSession session = request.getSession();
+    if (((int)session.getAttribute("isLoggedIn"))== 1) {
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/ProfileCreationServlet.java]: Please do

if (((int) session.getAttribute("isLoggedIn")) == 1) {

(missing spaces)

File: src/main/java/com/google/sps/servlets/ProfileCreationServlet.java [hunk 2]
```diff
@@ -38,33 +38,72 @@
...
+  }
+
+  public void createProfile(
+    HttpServletRequest request,
+    HttpServletResponse response,
+    HttpSession session
+   )  throws IOException {
+    String username = request.getParameter("username");
+    String bio = request.getParameter("bio");
+    String email = (String) session.getAttribute("uregisteredUserEmail");
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/ProfileCreationServlet.java]: typo, "unregistered"

File: src/main/java/com/google/sps/servlets/ProfileCreationServlet.java [hunk 3]
```diff
@@ -38,33 +38,72 @@
...
+    HttpServletRequest request,
+    HttpServletResponse response,
+    HttpSession session
+   )  throws IOException {
+    String username = request.getParameter("username");
+    String bio = request.getParameter("bio");
+    String email = "";
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    long id = (long) session.getAttribute("id");
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/ProfileCreationServlet.java]: Move this up with the rest of the parameters. Rename it to "userId".

File: src/main/java/com/google/sps/servlets/ProfileCreationServlet.java [hunk 4]
```diff
@@ -38,33 +38,72 @@
...
+    Entity userEntity = null;
+    try {
+      userEntity = getUserById(datastore, id);
+      email = (String) userEntity.getProperty("email");
+    } catch (EntityNotFoundException e) {
+      request.setAttribute("error", 1);
```
Comment (ramaroberto) [inline src/main/java/com/google/sps/servlets/ProfileCreationServlet.java:103]: If you don't need anything else from the lines below, I'd do 

request.setAttribute("error", 1);
response.sendRedirect("/user?id=" + id);
return;

Here.

**Review by ramaroberto — APPROVED** [review]

**Pull request merged**